src/datasets/arrow_dataset.py

```diff
@@ -3029,6 +3032,135 @@ def select(
         if len(self) == 0:
             return self
 
+        # If the indices are contiguous, simply slice the arrow table
+        if isinstance(indices, range):
+            if _is_range_contiguous(indices) and indices.start >= 0:
+                start, length = indices.start, indices.stop - indices.start
+                return self._select_contiguous(start, length, new_fingerprint=new_fingerprint)
+        else:
+            try:
+                start = next(iter(indices))
+            except StopIteration:
+                return self._select_contiguous(0, 0, new_fingerprint=new_fingerprint)
```

Comment:
Naive question: which use case is this for?

Reply:
It's in case `indices` is an empty iterable